application/src/main/java/org/togetherjava/tjbot/features/help/PinAnswerCommand.java

```diff
@@ -0,0 +1,73 @@
+package org.togetherjava.tjbot.features.help;
+
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.entities.channel.concrete.ThreadChannel;
+import net.dv8tion.jda.api.events.interaction.command.MessageContextInteractionEvent;
+import net.dv8tion.jda.api.interactions.commands.build.Commands;
+
+import org.togetherjava.tjbot.features.BotCommandAdapter;
+import org.togetherjava.tjbot.features.CommandVisibility;
+import org.togetherjava.tjbot.features.MessageContextCommand;
+
+public final class PinAnswerCommand extends BotCommandAdapter implements MessageContextCommand {
+    private static final String COMMAND_NAME = "pin-answer";
+    private static final int MAX_PINNED_ANSWERS = 10;
+    private int count = 0;
+
+    public PinAnswerCommand() {
+        super(Commands.message(COMMAND_NAME), CommandVisibility.GUILD);
+    }
+
+    @Override
+    public void onMessageContext(MessageContextInteractionEvent event) {
+        Message originalMessage = event.getTarget();
+        User commandInvoker = event.getUser();
+
+        if (!(originalMessage.getChannel() instanceof ThreadChannel threadChannel)) {
+            replyNotInThread(event);
+            return;
+        }
+
+        if (!isThreadCreator(commandInvoker, threadChannel)) {
+            replyNotThreadCreator(event);
+            return;
+        }
+
+        if (count >= MAX_PINNED_ANSWERS) {
+            replyMaxPinsReached(event);
+            return;
+        }
```

Comment:
`count` might not be reliable to keep track of pinned threads so far, 
* what happens when bot is offline for whatever reasons?
* what about if a mod pins a new message in that channel that's not done via this command?

So in this case it's better to rely on discord to stay upto date with current state of thread
[retrievePinnedMessages()](https://javadoc.io/doc/net.dv8tion/JDA/latest/net/dv8tion/jda/api/entities/channel/middleman/MessageChannel.html#retrievePinnedMessages()) might be more helpful. Get the list, find out and then verify if you can pin a new message or not.